solutions/usaco-1011.mdx

```diff
@@ -95,4 +95,58 @@ public class triangles{
 
 </JavaSection>
 
+<CPPSection>
+
+```CPP
+
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	freopen("triangles.in", "r", stdin);
+	freopen("triangles.out", "w", stdout);
+	int n; cin >> n;
+	int x[n];
+	int y[n];
+	int ans = 0;
+	for(int i = 0; i < n; i++) {
+		cin >> x[i] >> y[i];
+	}
+	for(int i = 0; i < n; i++) {
+		for(int j = 0; j < n; j++) {
+			for(int k = 0; k < n; k++) {
+				// Check to see if both points are parallel to the X and Y axis
+				if(y[i] ==y [j] && x[i] == x[k]) {
+					/** 
+					 * Since we need to output two times the
+					 * area and the formula for the area
+					 * of a triangle is 1/2 * base * height, 
+					 * we can omit the 1/2 because it anyways
+					 * gets multiplied by 2, which cancels it out. 
+					 * Finding the base is the difference
+					 * between both x points and the height is 
+					 * the difference between both y points;
+					 * it is a right triangle because both 
+					 * sides are parallel to the x and y axis,
+					 * so no advanced math is required to find 
+					 * the height.
```

Comment:
this could definently be shortened, something like "The problem asks us to output two times the area, so the triangle formula of 1/2bh turns into bh..."

Reply:
i wanna add on to this- comments in the code are just supposed to give a quick explanation- they aren't supposed to be this long (at least imo)
if you feel like this really needs a long explanation, put it in the editorial and say something like "does [x] as explained in the editorial"

Reply:
I agree I dont think i saw the editorial that the others wrote and I think what they wrote is good enough so imma just delete this part.